src/components/crossDevice/MobileFlow.js

```diff
@@ -0,0 +1,38 @@
+import { h, Component } from 'preact'
+
+import MobileConnected from './MobileConnected'
+import MobileComplete from './MobileComplete'
+
+class MobileFlow extends Component {
+  componentDidMount() {
+    this.props.socket.on('get config', this.sendConfig)
+    this.props.socket.on('complete', this.onMobileComplete)
+  }
+
+  componentWillUnmount() {
+    this.props.socket.off('get config')
+    this.props.socket.off('complete')
+  }
+
+  sendConfig = (data) => {
+    if (this.props.roomId && this.props.roomId !== data.roomId) {
+      this.props.socket.emit('leave', {roomId: this.props.roomId})
+    }
+    this.props.actions.setRoomId(data.roomId)
+    this.sendMessage('config', this.props.mobileConfig, data.roomId)
+  }
+
+  sendMessage = (event, payload, roomId) => {
+    this.props.socket.emit('message', {event, payload, roomId})
+  }
+
+  onMobileComplete = () => {
+    this.props.actions.setMobileComplete(true)
+  }
+
+  render = (props) =>
+    this.props.mobileComplete ?
+      <MobileComplete/> : <MobileConnected back={props.back}/>
```

Comment:
Why are you using `back` and not `previousStep`?

Reply:
because previous step updates the history, which means if you were then to press back from the link screen you would go to the connected screen. If instead we trigger a back, if you were to then press back from the link screen you would got the the upload screen